Feature/equality transformer

Chip2916 opened the pull request:
Picked up second aspect of this ticket - https://github.com/lvgig/tubular/issues/204

Added an additional base class for two columns transformers and an associated test script which inherits from TwoColum Init in base tests. Might have missed the mark a little on this one, so have set up as a draft.

limlam96 commented:
Looks good! could you also:

- update changelog
- For any relevant transformers that have already been through the test refactor, could you update to inherit from this new base class (leave those that haven't been through the refactor for now as it will likely break lots of tests and cause a headache)

Chip2916 commented:
Hey @limlam96 - forgot changelog again!!! Updated now, and added inheritance from new base class to equality checker transformer (don't think date transformers have been refactored to use two cols yet)

limlam96 commented on tubular/comparison.py:
```diff
@@ -31,7 +31,7 @@ def __init__(
-        super().__init__(columns=columns, **kwargs)
+        super().__init__(columns=columns, new_col_name=new_col_name, **kwargs)
 
         if not (isinstance(columns, list)):
```
can now delete lines 36:46 :)
